app/pb_kits/playbook/pb_gauge/_gauge.jsx

```diff
@@ -0,0 +1,105 @@
+/* @flow */
+
+import React, { useEffect, useRef } from 'react'
+import { pbChart } from '../'
+import { spacing } from '../utilities/spacing.js'
+import classnames from 'classnames'
+import Highcharts from 'highcharts'
+
+import {
+  buildAriaProps,
+  buildCss,
+  buildDataProps,
+} from '../utilities/props'
+
+type GaugeProps = {
+  aria: Object,
+  className?: String,
+  chartData?: Array,
+  data?: Object,
+  fullCircle: Boolean,
+  height: String,
+  id?: String,
+  max: Number,
+  min: Number,
+  prefix: String,
+  showLabels: Boolean,
+  style: String,
+  suffix: String,
+  title: String,
+  tooltipHtml: String,
+}
+
+const Gauge = (props: GaugeProps) => {
+  const {
+    aria = {},
+    className,
+    chartData = [{ name: 'Name', value: 0 }],
+    data = {},
+    fullCircle = false,
+    height = null,
+    id,
+    max = 100,
+    min = 0,
+    prefix = '',
+    showLabels = false,
+    style = 'solidgauge',
+    suffix = '',
+    title = '',
+    tooltipHtml = '<span style="font-weight: bold; color:{point.color};">●</span>{point.name}: ' + '<b>{point.y}</b>',
+  } = props
+
+  const ariaProps = buildAriaProps(aria)
+  const dataProps = buildDataProps(data)
+
+  const css = buildCss({
+    'pb_gauge_kit': true,
+  })
+  // Runs first time component Renders
+  useEffect(() => {
+    chartData.forEach((obj) => {
+      obj.y = obj.value
+      delete obj.value
+    })
```

Comment:
Since chartData is coming as a prop, I think it's better if you transform this collection into another one

Reply:
I vaguely remember trying to use .map() and assigning it a new variable like `formattedChartData`.  Seem to remember it not working for whatever reason but let me give it another try.  Have made some structural changes since then so whatever the issue was might have resolved itself.

Reply:
Forgot to return the object after reformatting it inside `.map()`.  Thanks for drawing my attention back to this one, feels good to have that mystery resolved.  I will push this fix back up later this week.  Don't want to reset the Milano build, I have it linked as a doc reference for others reviewing the alpha.

Reply:
Why would we want that? I was actually thinking we could use the incoming `chartData` to determine when rerenders are required.

Reply:
Looks like `chartData` still determines rerenders, it will just be reformatted as it's passed in the first time.  Highcharts is particular (and not very semantic) about how data has to be formatted.  Subsequent renders will pass it's value field directly to the highcharts api to [update the chart's data](https://github.com/powerhome/playbook/blob/9c853a9c306b453deab4d0c1c4943d5c8999d4fe/app/pb_kits/playbook/pb_gauge/_gauge.jsx#L88).

Reply:
got it 👍🏼

Reply:
This has got me thinking though, might there be cases where you'd like to change not just the value in chartData but also its' name?  i.e. `chartData={[{name: 'init name', value: 'init value'}]}` mutating to `chartData={[{name: 'modified name', value: 'modified value'}]}`.  Right now I'm only accounting for changes in value, not changes to name.

Reply:
Updated functionality to include concerns in the comment above.  Now includes the ability to change name as well as value.

Reply:
yeah, that's hard to tell without an actual use-case... With that said, given that React components in general, are functions of `data` to `UI`, e.g. `fn: Props -> JSX`, I'd expect that a different set of props would give me a different UI state.